pkg/cmd/experimental/diagnostics/diagnostics.go

```diff
@@ -43,141 +71,148 @@ can be used with subcommands.
 `
 
 func NewCommandDiagnostics(name string, fullName string, out io.Writer) *cobra.Command {
-	opts := options.NewAllDiagnosticsOptions(out)
+	o := &OverallDiagnosticsOptions{
+		RequestedDiagnostics: AvailableOverallDiagnostics.List(),
+		LogOptions:           &log.LoggerOptions{Out: out},
+	}
+
 	cmd := &cobra.Command{
 		Use:   name,
 		Short: "This utility helps you understand and troubleshoot OpenShift v3.",
 		Long:  fmt.Sprintf(longAllDescription, fullName),
 		Run: func(c *cobra.Command, args []string) {
-			opts.GlobalFlags = c.PersistentFlags()
-			run.Diagnose(opts)
+			kcmdutil.CheckErr(o.Complete())
```

Comment:
I really don't ever want the user to see a stack trace or glog output from this, no matter what they (or I) have screwed up.